src/mender-update/daemon/states.cpp

```diff
@@ -103,15 +103,59 @@ void IdleState::OnEnter(Context &ctx, sm::EventPoster<StateEvent> &poster) {
 	log::Debug("Entering Idle state");
 }
 
+SubmitInventoryState::SubmitInventoryState(
+	events::EventLoop &event_loop, int retry_interval_seconds, int retry_count) :
+	retry_(Retry {
+		http::ExponentialBackoff(chrono::seconds(retry_interval_seconds), retry_count),
+		event_loop}) {
+}
+
+void SubmitInventoryState::HandlePollingError(Context &ctx, sm::EventPoster<StateEvent> &poster) {
+	// On shorter polling intervals, we need to adjust the backoff to not set
+	// intervals longer than the maximum retry polling interval
+	chrono::milliseconds max_interval =
+		chrono::seconds(ctx.mender_context.GetConfig().retry_poll_interval_seconds);
+	if (max_interval < retry_.backoff.SmallestInterval()) {
+		retry_.backoff.SetSmallestInterval(max_interval);
+		retry_.backoff.SetMaxInterval(max_interval);
```

Comment:
This turns the exponential backoff into a fixed one, right? Worth a comment, I believe.

Reply:
Yes, it does. I'll add a short comment :+1: